### PR TITLE
BUGFIX: Fix mediabrowser regressions, improve configuration

### LINKED ISSUE #1284: Move translation for the Media package
Now that the Media package as a dedicated package, the translation needs to be moved to the new package.

How to take care of the available translation in other languages ?

### PR DESCRIPTION
**Pull request by Sebobo**
This acts on feedback after merge of #159 

Closes #1284 

# Things to be worked on

- [x] Add useful description to package
- [x] Clean up the two AssetController classes
- [x] Auto include routes
- [x] Fix requestPatterns regression
- [X] Fix whitespace regression document breadcrumb
- [x] Move translation labels to media browser package
- [x] Move the stylesheets to the new package
- [x] Move thumbnail configuration the Neos.Neos
- [x] Upload broken is user select a tag before upload (affect only master, confirmed by @gerhard-boden see https://neos-project.slack.com/archives/team-unicorn/p1481721419000067)
- [x] Drag'n drop to Collection / Tag don't work

# Infrastructure

- [x] Sync CrowdIn (translation move, see https://github.com/neos/neos-development-collection/pull/1305/commits/94ff76913907cfe80743ecbe9e11591de5b4e846)
- [x] Update Jenkins build script to build CSS/JS

### CODE REVIEW

**Comment by dfeyer** [comment]
Hi @Sebobo why this is WIP and where did you need help ?

**Comment by dfeyer** [comment]
Tagging assets return an error

**Comment by dfeyer** [comment]
Issue with translations

**Comment by Sebobo** [comment]
@dfeyer I started working on the feedback of Aske in #159. But I don't have a good idea or knowledge for everything as I was not involved in the initial implemenation of the change.
There are some issues left. Would be great if could check the feedback at the end of #159 and we can then discuss who of us does what and how :)

**Comment by dfeyer** [comment]
@Sebobo OK so look like a lots of places need some attentions :)

**Comment by Sebobo** [comment]
@dfeyer I updated the description with a todo list

**Comment by dfeyer** [comment]
Translation are moved and flash message are now displayed correctly, but wait @kdambekalns feedback on how to sync with CrowdIn

**Comment by dfeyer** [comment]
Ok I continue on this one this afternoon

**Comment by dfeyer** [comment]
@aertmann Can you give us more informations about the regression with the Policy (see https://github.com/neos/neos-development-collection/pull/159#issuecomment-263950109) ?

**Comment by dfeyer** [comment]
> Move thumbnail generation and configuration

With https://github.com/neos/neos-development-collection/pull/1305/commits/b97140ddbdffe3987a4d809abe3774844a3e55ca the thumbnail preset configuration are back the the Neos package. Not sure what you mean by "generation" ?

**Comment by Sebobo** [comment]
@dfeyer I wasn't sure where the ThumbnailGenerator class should be. But if the Media Package is fine and the naming matches then no change is needed regarding that.

**Comment by dfeyer** [comment]
> I wasn't sure where the ThumbnailGenerator class should be. But if the Media Package is fine and the naming matches then no change is needed regarding that.

@Sebobo OK so from my POV, currently the Neos.Media package is "coupled" with Neos.Neos only because we auto load the routing "before Neos.Neos" ... and that's bad. This package should depend on Flow only. So ThumbnailGenerator must be a service provided by Neos.Neos or Neos.Media.Browser ... but more a Neos service, because different implementation of the Browser can use the same async thumbnail génération, and maybe that can even be used in frontend at some point or somewhere else in the UI. @aertmann did you agree ?

About the tagging regression it affection create and edit action, it's really a regression as everything work fine in the LTS version. Let see I we can solve this one.

**Comment by dfeyer** [comment]
> Having two layout templates is a bit messy, but maybe not solveable for now.

**Comment by dfeyer** [comment]
By the way we have to move the Stylesheet also to this package, I update the todo

**Comment by dfeyer** [comment]
You need to build the Stylesheet to test the last version (Neos.Neos + Neos.Media.Browser)

In the root directory of the package

    bunder install
    npm install
    grunt build

**Comment by dfeyer** [comment]
I revert the cleanup of Views.yaml because it break the secondary inspector

**Comment by dfeyer** [comment]
So I think this one is ready for review, but don't merge it before the checklist under "Infrastructure" is done done.

Regarding: "Policy configuration regression (not sure where)", please @aertmann can you give us more input I don't see any regression in this area ?

**Comment by dfeyer** [comment]
The points describe in "Nice to have" can be solved later one.

**Comment by dfeyer** [comment]
Added "Drag'n drop to Collection / Tag don't work" to the todo list

**Comment by dfeyer** [comment]
https://github.com/neos/neos-development-collection/pull/1305/commits/d5824fd4097bb658d22d0abc633ce68341c735c1 remove all translation, need to found a proper way to move migration for "media.*" from Neos the Media.Browser package.

**Comment by robertlemke** [comment]
@dfeyer please let us know when you think that this PR is finished / mergeable so we can do a final review.

**Comment by Sebobo** [comment]
@robertlemke the drag & drop issue is fixed. Now the Jenkins build script change is needed and we are good to go I think.

**Comment by dfeyer** [comment]
> @robertlemke the drag & drop issue is fixed. Now the Jenkins build script change is needed and we are good to go I think. Will continue testing.

Yes basically CrowdIn + Jenkins build and it's merge ready ... I'm sick ... last day at the office and holiday until tomorrow, so unable to help during the next two weeks ... and perfect time to be sick ...

**Comment by aertmann** [comment]
Hey the policy regression was related to the request patterns setting regression.. sorry for the confusion :/

**Comment by robertlemke** [comment]
All right, do we have +1s for this patch (tested?) or should we still give this a thorough review before merging it? If so, we supposedly can't merge it today before the Developer Preview announcement, but keep it as a known issue.

**Comment by dfeyer** [comment]
This one is ready for review, one small issue with the sticky for in edit action, when you scroll the sticky bar at the bottom is not sticky anymore sometime (chrome browser), @aertmann maybe you have a better idea how to solve that ?

And other issue when accessing the RelatedNodes action from the Edit action, sometimes the template does not render, but it's not related the this PR, I discuss the topic with @kitsunet 

Remember to test this PR:
- Build the CSS from the Media Browser package
- Rebuild the CSS of Neos (to remove the media browser styles)

Please review guys

**Comment by dfeyer** [comment]
@aertmann Look like it's a issue with recent chrome and fixedsticky.js

Regarding the support in http://caniuse.com/#feat=css-fixed I think we can simple remove the fixedsticky.js and raise the IE minimum version to 11

**Comment by dfeyer** [comment]
@kdambekalns Any clue what we can do with CrowdIn translation ?

And last question how did we handle the CSS compilation (Jenkins, commited on git, ...) ?

**Comment by kdambekalns** [comment]
Answer to the last question first: by now we have https://jenkins.neos.io/job/neos-build-minified/ and https://jenkins.neos.io/job/neos-build-minified-2.x/ for the JS/CSS build. So that should just work.

As far as the translation goes, that needs an update to a not-yet-created job similar to https://jenkins.neos.io/job/crowdin-sync-2.3/. Right now that means, just go ahead. The moved files will be uploaded once I finally create that job.

**Comment by aertmann** [comment]
Hey started reviewing this one and found a bunch of things and ended up spending 12 hours on it 😕

@Sebobo: Could you merge https://github.com/Sebobo/neos-development-collection/pull/1 into this one and then review it again? Thanks

**Comment by Sebobo** [comment]
@aertmann wow, that's quite a bit of stuff.
I merged it and will later check it out. We really have to get this done at some point...

File: Neos.Media.Browser/Classes/Domain/Session/BrowserState.php [hunk 1]
```diff
@@ -27,7 +27,7 @@ class BrowserState
         'activeTag' => null,
         'view' => 'Thumbnail',
         'sortBy' => 'Modified',
-        'sortDirection' => 'ASC',
+        'sortDirection' => 'DESC',
```
Comment (aertmann) [inline Neos.Media.Browser/Classes/Domain/Session/BrowserState.php]: same as https://github.com/neos/neos-development-collection/pull/1392

**Comment by aertmann** [comment]
@Sebobo: yeah a bit of a doozy, turned into a little monster

just noticed two more things, will push fixes

thanks for merging it

**Pull request merged**